### ISSUE TITLE
Методы ЗначениеРеквизитаОбъекта и ЗначенияРеквизитовОбъекта

### ISSUE THREAD

**Issue by KonstantinHeinrich** [SeiOkami/OneS#28]
В БСП в модуле `ОбщегоНазначения `есть самые популярные во всех типовых конфигурациях методы ЗначениеРеквизитаОбъекта и ЗначенияРеквизитовОбъекта, что подтверждено [недавней статистикой](https://infostart.ru/1c/articles/1398340).
Оно и понятно, ведь эти методы избавляют от обращения к реквизитам ссылки и чтения всего объекта с возможной блокировкой, т.е. неоптимальной работы кода. Однако многие не используют эти методы, просто потому что не знают про них или они просто громоздкие.
Предлагаю перенести эти функции в платформу как методы ссылки. Сравните вызов:
`ОбщегоНазначения.ЗначениеРеквизитаОбъекта(МояСсылка, «МойРеквизит»)`
И возможный:
`МояСсылка.ЗначениеРеквизита(«МойРеквизит»)`
Какие плюсы:
1. Запись намного короче
2. Работает контекстная подсказка
3. Не требуется БСП

Т.к. ссылка доступна на клиенте, то и эти методы можно сделать доступными с клиента, конечно, с вызовом сервера. Тогда не надо будет на формах создавать отдельную серверную функцию ил искать модуль `ОбщегоНазначения…ВызовСервера`.
В идеале еще сделать подсказку по реквизитам в параметре-строке. Подобные подсказки в строках уже есть, например, для методов `ОткрытьФорму` или `ПредопределенноеЗначение`.

**Comment by emiliskan** [comment]
зачем вообще функция, лучше если платформа сразу при получении одного реквизита будет оборачивать это все

**Comment by KonstantinHeinrich** [comment]
> зачем вообще функция, лучше если платформа сразу при получении одного реквизита будет оборачивать это все

Нет, это уже будет слишком, есть механизм получения реквизитов ссылки, который существует уже как 20 лет. Ведь происходит еще кэширование, и при повторном обращении к другому реквизиту вызова БД уже не будет. Согласен, что это кейс намного более редкий, нежели единичное получение реквизита, но это будет полная смена логики.

**Comment by VVi3ard** [comment]
Тогда уж лучше сделать в виде Контрагент.Реквизиты("Наименование", "Код").Получить()
тогда можно будет делать так:
РеквизитыДляЧтения =  Контрагент.Реквизиты("Наименование");
....
РеквизитыДляЧтения.Добавить("Адрес");
....
РеквизитыДляЧтения.Прочитать();

Плюс такой вызов можно будет и кешировать нормально (через версию объекта). 
Естественно должен быть конструктор на основании массива и структуры.

**Comment by andrewzola** [comment]
Еще лучше
МояСсылка:МойРеквизит

**Comment by emiliskan** [comment]
> Еще лучше
> 
> МояСсылка:МойРеквизит

Зачем так

**Comment by andrewzola** [comment]
короче и удобнее запись

**Comment by VVi3ard** [comment]
А если нужно сразу несколько реквизитов и часть из них через разыменование?
Если в начале вы не знаете какие именно понадобятся и хотели бы по ходу алгоритма готовить список для запроса?
А если список у вас хранится в масссиве? Или в структуре?

**Comment by KonstantinHeinrich** [comment]
> А если нужно сразу несколько реквизитов и часть из них через разыменование?
> Если в начале вы не знаете какие именно понадобятся и хотели бы по ходу алгоритма готовить список для запроса?
> А если список у вас хранится в масссиве? Или в структуре?

Все это поддерживает БСПшный метод `ЗначенияРеквизитовОбъекта`. Его функционал и перенести.

**Comment by marmyshev** [comment]
Не очень понял предложение с подсказками - вроде бы это уже всё сделано в SSL-support

 https://github.com/1C-Company/ssl-support/blob/master/docs/common-object-attribute-value.md

**Comment by KonstantinHeinrich** [comment]
> Не очень понял предложение с подсказками - вроде бы это уже всё сделано в SSL-support
> 
> https://github.com/1C-Company/ssl-support/blob/master/docs/common-object-attribute-value.md

Это круто, но чтобы это работало надо использовать EDT и поставить плагин.
И все равно это решит только пункты 2 и 3 из ишузы.

**Comment by SeiOkami** [comment]
Отправлено боту 1С 06.04.2021 в 20:03
